oteapi/models/transformationconfig.py

```diff
@@ -29,7 +30,7 @@ class ProcessPriority(str, Enum):
     HIGH = "High"
 
 
-class TransformationConfig(GenericConfig):
+class TransformationConfig(GenericConfig, SecretConfig):  # type: ignore [misc]
```

Comment:
I need to check locally why this mypy ignore rule is necessary and whether it can be "fixed". This is true for all of the ignore statements in this case.

Reply:
So this seems to be related to the other unresolved comment here.
I found this rather lengthy, detailed, and interesting discussion on the topic: python/mypy#1237.

As a result, I tried to workaround this "issue" and found a solution I'm happy with:

Don't use `AttrDict` as a base class for `SecretConfig`. There is no need.
Then, additionally, set the `json_dumps` configuration as a class keyword argument (see the 2nd code block in the pydantic documentation [here](https://docs.pydantic.dev/usage/model_config/)).

Now you can remove all the `# type: ignore` lines you've added, also update the type for `model` in the `json_dumps()` method to `dict` and all should be well.
I have done this locally and can push the commit if you want? Otherwise, feel free to do it yourself as well :)